### PR TITLE
Branch for test

### PR DESCRIPTION
**Pull request by SerezhkaGit**

### CODE REVIEW

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Your code coverage should not be less than 80% of covered lines.

To test your code coverage locally - run mvn clean verify in the terminal.
Provide test for all neccessary services and classes

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Provide screenshot with test coverage
Create test classes and tests for checking the work of each of your services

**Comment by SerezhkaGit** [comment]
image

**Comment by SerezhkaGit** [comment]
image

File: src/test/java/core/basesyntax/MainTests.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import core.basesyntax.service.impl.ReportGeneratorImpl;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+public class MainTests {
+    private final FileReaderImpl fileReader = new FileReaderImpl();
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/MainTests.java]: initialise in `@BeforeEach` method
same for other places

File: src/test/java/core/basesyntax/FruitTransactionTest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import org.junit.jupiter.api.Test;
+
+public class FruitTransactionTest {
+    @Test
+    void constructorAndGetters_ShouldWorkCorrectly() {
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/FruitTransactionTest.java]: don't test Java doc

File: src/test/java/core/basesyntax/MainTests.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    void read_ShouldReadFileSuccessufully() {
+        String fileName = "reportToRead.csv";
+        List<String> lines = fileReader.read(fileName);
+        assertNotNull(lines);
+        assertFalse(lines.isEmpty());
+        assertEquals("type,fruit,quantity", lines.get(0));
+    }
+
+    @Test
+    void main_ShouldRunWithoutExceptions() {
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/MainTests.java]: remove

File: src/test/java/core/basesyntax/MainTests.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+import core.basesyntax.service.impl.FileWriterImpl;
+import core.basesyntax.service.impl.ReportGeneratorImpl;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+public class MainTests {
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/MainTests.java]: each test should be responsible for particular logic / service
You should have 
```
FileReaderTest
FileWriterTest
ReportGeneratorTest
```
etc

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Comment by SerezhkaGit** [comment]
image

File: src/test/java/core/basesyntax/FileReaderImplTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        List<String> lines = Files.readAllLines(tempFile);
+
+        assertEquals(1, lines.size());
+        assertEquals("line1", lines.get(0));
+
+        Files.deleteIfExists(tempFile);
+    }
+
+    @Test
+    void read_ShouldThrowException_WhenFileDoesNotExist() {
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/FileReaderImplTest.java:37]: add test case when file is empty

File: src/test/java/core/basesyntax/FruitTransactionTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import org.junit.jupiter.api.Test;
+
+public class FruitTransactionTest {
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/FruitTransactionTest.java]: remove test

File: src/test/java/core/basesyntax/ReportGeneratorImplTest.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import core.basesyntax.service.impl.ReportGeneratorImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ReportGeneratorImplTest {
+    private ReportGenerator reportGenerator;
+
+    @BeforeEach
+    void setUp() {
+        Storage.fruits.clear();
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/ReportGeneratorImplTest.java]: clear storage in `@AfterEach` method

File: src/test/java/core/basesyntax/ShopServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ShopServiceImplTest {
+    private ShopService shopService;
+
+    @BeforeEach
+    void setUp() {
+        Storage.fruits.clear();
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/ShopServiceImplTest.java]: same here
check other places

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]
